### PR TITLE
Create chart api and components

### PR DESCRIPTION
**Pull request by luqven**
### **Pending**
~~- [ ]  Create a skeleton for a graph component~~
~~- [ ]  Create a Skeleton for a watchlist component that uses many Graph components~~
### **Done**
- [x]  Create Stocks / Crypto API that fetches real-time data
---
#### Description

This commit uses the [AlphaVantage](https://www.alphavantage.co/) API to fetch real-time stock data. The JavaScript wrapper for the API came from [zackurben](https://github.com/zackurben/alphavantage), which also comes with a neat normalizing function since the default output JSON is so ugly.

Right now, most of the changes are centered around the `home_page.jsx` component. There, there is a hardcoded button to make a request for `MSFT` quote data. This is meant to demonstrate:
- how the API can be utilized
- ~~how to use `useReducer` (from [react-hooks](https://reactjs.org/docs/hooks-reference.html#usereducer)) instead of Redux for these types of calls~~
- Turns out that `useReducer` does not play nice with `redux`.  We're better off sticking to `useState`. Oh, and `react-hooks` also break _storybook_, so there's another reason not to use `useRedecer`.

### CODE REVIEW

**Review by guyeilam — APPROVED** [review]

File: frontend/src/components/chart/watch_list.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { useState } from "react";
+// import styled from "styled-components";
+import Button from "../library/styled_button";
+import { fetchQuote } from "../../actions/alphavnatage_actions";
+
+export default function WatchList(props) {
+  //  react-hook reducer
+  const [data, setData] = useState({ data: {} });
+
+  const handleSumit = () => {
```
Comment (guyeilam) [inline frontend/src/components/chart/watch_list.jsx]: Typo: handleSubmit

File: frontend/src/components/chart/watch_list.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    props
+      .fetchQuote("MSFT", "compact", "json", "60min")
+      .then(result => setData({ data: result }));
+  };
+
+  return (
+    <div>
+      {/* <p>{Object.values(data.data)}</p> */}
+      <Button submit={handleSumit}>Fetch MSFT Quote</Button>
```
Comment (guyeilam) [inline frontend/src/components/chart/watch_list.jsx]: Typo: handleSubmit

File: frontend/src/components/chart/watch_list_container.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { connect } from "react-redux";
+import WatchList from "./watch_list";
+import { fetchQuote } from "../../actions/alphavnatage_actions";
```
Comment (guyeilam) [inline frontend/src/components/chart/watch_list_container.js]: Typo: alphavantage_actions (also fix typo in the actual file name)

File: frontend/src/reducers/alpha_reducer.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import {
+  RECEIVE_INTRADAY,
+  RECEIVE_BATCH,
+  RECEIVE_QUOTE,
+  RECEIVE_RATE,
+  RECEIVE_DAILY,
+  RECEIVE_WEEKLY,
+  RECEIVE_MONTHLY
+} from "../actions/alphavnatage_actions";
```
Comment (guyeilam) [inline frontend/src/reducers/alpha_reducer.js]: Fix typo (alphavantage_actions)

File: frontend/src/util/alphavantage_keys.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+const apiKey = "7XF2MDOK9MZJBHFJ";
```
Comment (guyeilam) [inline frontend/src/util/alphavantage_keys.js:1]: This file should probably be added to gitignore

Comment (luqven) [reply frontend/src/util/alphavantage_keys.js:1]: lol yes rather urgently too 😱 good catch!

**Pull request merged**